src/client/Item/Item.ts

```diff
@@ -32,15 +32,6 @@ export class Item {
     private readonly clientContext: ClientContext
   ) {}
 
-  /**
-   * Read the item's definition.
-   *
-   * There is no set schema for JSON items. They may contain any number of custom properties.
-   *
-   * @param options Additional options for the request, such as the partition key.
-   * Note, if you provide a partition key on the options object, it will override the primary key on `this.partitionKey`.
-   */
-  public read(options?: RequestOptions): Promise<ItemResponse<ItemDefinition>>;
```

Comment:
Why you delete the non-T version of this method? I'm open to that change, but seems like it's increasing the scope of what this PR is trying to do.

Reply:
I've been removing these and adding default types as I go. I think they can all use default types, so decreases future maintenance load.